dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/shell/AbstractShell.java

```diff
@@ -281,7 +283,7 @@ public void run() {
         //Process has not terminated.
         //So check if it has completed 
         //if not just destroy it.
-        if (p != null && !shell.completed.get()) {
+        if (!shell.completed.get()) {
           shell.setTimedOut();
```

Comment:
If  runCommand is not called, process may  be null , use ` !(p  instanceof Process) ` to check it.

Reply:
process vaiable will be assigned before ShellTimeoutTimerTask, so no need to check